model/src/pyrenew/process/simplerandomwalk.py

```diff
@@ -62,12 +63,20 @@ def sample(
 
         if init is None:
             init = npro.sample(name + "_init", self.error_distribution)
-        diffs = npro.sample(
-            name + "_diffs",
-            self.error_distribution.expand((n_timepoints - 1,)),
+
+        def transition(x_prev, _):
+            # numpydoc ignore=GL08
+            diff = npro.sample(name + "_diffs", self.error_distribution)
```

Comment:
What gets tracked by this `sample` call when inference is performed? Is only the final value in the scan logged? Is a vector of length `n_timepoints - 1` created?

Reply:
`Rt_transformed_rw_diffs[0]`, `Rt_transformed_rw_diffs[1]`, … `Rt_transformed_rw_diffs[n]` are all created.

Reply:
Excellent.